image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -296,28 +294,28 @@ def main(args):
 
         cmds_run(['echo "Uninstalling packages"'])
         uninstall_packages(argsDict['ImageName'][0], argsDict['Version'][0])
-
+        
+        
+        
         cmds_run(['echo "Creating Repo File"'])
         if 'RepoName' in argsDict:
-            create_repo_file(argsDict['RepoName'][0], argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0])
+            copy_gpg(argsDict['ImageName'][0], workingDir, argsDict['RpmPublicKey'][0])
+            create_repo_file(argsDict['RepoName'][0], argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0],"/tmp/"+argsDict['RpmPublicKey'][0])
         else:
-            create_repo_file('Nuage', argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0])
+            copy_gpg(argsDict['ImageName'][0], workingDir, argsDict['RpmPublicKey'][0])
+            create_repo_file('Nuage', argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0],"/tmp/"+argsDict['RpmPublicKey'][0])
 
         cmds_run(['echo "Installing Nuage Packages"'])
         install_packages(argsDict['ImageName'][0])
 
-        cmds_run(['echo "Installing VRS"'])
-        install_vrs(argsDict['ImageName'][0])
-
         cmds_run(['echo "Cleaning up"'])
-        if 'RepoName' in argsDict:
-            delete_repo_file(argsDict['RepoName'][0], argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0])
-        else:
-            delete_repo_file('Nuage', argsDict['RepoBaseUrl'][0], argsDict['ImageName'][0])
 
+        delete_repo_file(argsDict['ImageName'][0])
+        delete_gpg(argsDict['ImageName'][0], argsDict['RpmPublicKey'][0])
 
         if 'AVRSBaseUrl' in argsDict:
-            create_repo_file('6wind', argsDict['AVRSBaseUrl'][0], argsDict['ImageName'][0])
+            copy_gpg(argsDict['ImageName'][0], workingDir, argsDict['RpmPublicKey'][0])
```

Comment:
For all the Nuage packages, there is only 1 gpgkey. So, let's add gpgkey in line 299 and remove it after line 328.